src/components/smart_objects/include/smart_objects/number_schema_item.h

```diff
@@ -157,11 +157,20 @@ errors::eType TNumberSchemaItem<NumberType>::validate(
     report->set_validation_info(validation_info);
     return errors::INVALID_VALUE;
   }
+
   NumberType value(0);
   if (typeid(int32_t) == typeid(value)) {
-    value = utils::SafeStaticCast<int64_t, int32_t>(Object.asInt());
+    if (Object.asInt() > std::numeric_limits<int32_t>::max() ||
+        Object.asInt() < std::numeric_limits<int32_t>::min()) {
```

Comment:
One issue with this approach, some integer parameters don't have a `maxvalue` param, so those should be able to handle whatever value is sent (barring it being larger that 64 bits, since that is a hard limit).
Something like `cancelID` is currently stored in a 32 bit integer, but has no limitations as far as range. This change would make it so any value larger than 32 bits would be rejected.

To solve this, I think that we should use 64 bits for parameters which are missing either `maxValue` or `minValue`. This would need to be done in the RPC spec parser script. That, combined with this fix, should cover all cases.

Reply:
@jacobkeeler generator script has been updated in https://github.com/smartdevicelink/sdl_core/pull/3750/commits/2852ac4f1c79432194480c159dfb1a5d55d154fa and https://github.com/smartdevicelink/sdl_core/pull/3750/commits/9891d9398d2bf72c37043c538e3b3eb04e33b8da
Now generator has rules for int/uint 32/64 by analyzing `minValue` and `maxValue` and providing the best fit for each integer in the schema.
Note that schema item itself has been updated to properly reflect all changes from generator.